common/src/main/java/com/codeforcommunity/auth/Passwords.java

```diff
@@ -14,76 +13,114 @@
  * still considered robust and <a href="https://security.stackexchange.com/a/6415/12614"> recommended by NIST </a>.
  * The hashed value has 256 bits.
  */
-public class Passwords {
-
+public final class Passwords {
   private static final Random RANDOM = new SecureRandom();
+  // Number of iterations to run for hashing the password.
   private static final int ITERATIONS = 10000;
+  // Key length for the hash.
   private static final int KEY_LENGTH = 256;
+  // Salt length for the hash.
+  private static final int SALT_LENGTH = 16;
+  // Secret key algorithm.
   private static final String SECRET_KEY_DERIVATION = "PBKDF2WithHmacSHA1";
 
   /**
    * Returns a random salt to be used to hash a password.
    *
-   * @return a 16 bytes random salt
+   * @param length length of salt to be returned.
+   *
+   * @return a bytes random salt.
    */
-  public static byte[] getNextSalt(int length) {
+  private static byte[] getNextSalt(int length) {
     byte[] salt = new byte[length];
     RANDOM.nextBytes(salt);
     return salt;
   }
 
-  public static byte[] getNextSalt() {
-    return getNextSalt(50);
+  /**
+   * Returns a random salt of SALT_LENGTH size.
+   *
+   * @return a 16 byte random salt.
+   */
+  private static byte[] getNextSalt() {
```

Comment:
While I appreciate the thought behind the auxilary function I think because this method is private and we only use it in one place it keeps things cleaner if we just stick with the one version

Reply:
ugh fine